src/systems/particle_emitter/ParticleEmitter.cc

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/msgs/particle_emitter.pb.h>
+
+#include <string>
+
+#include <ignition/common/Profiler.hh>
+#include <ignition/msgs/Utility.hh>
+#include <ignition/plugin/Register.hh>
+#include <sdf/Material.hh>
+
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParticleEmitter.hh>
+#include <ignition/gazebo/components/Pose.hh>
+#include <ignition/gazebo/Conversions.hh>
+#include "ParticleEmitter.hh"
+
+using namespace ignition;
+using namespace gazebo;
+using namespace systems;
+
+// Private data class.
+class ignition::gazebo::systems::ParticleEmitterPrivate
+{
+  /// \brief The particle emitter.
+  public: ignition::msgs::ParticleEmitter emitter;
+};
+
+//////////////////////////////////////////////////
+ParticleEmitter::ParticleEmitter()
+  : System(), dataPtr(std::make_unique<ParticleEmitterPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void ParticleEmitter::Configure(const Entity &/*_entity*/,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    EntityComponentManager &_ecm,
+    EventManager &/*_eventMgr*/)
+{
+  // Name.
+  if (!_sdf->HasElement("emitter_name"))
```

Comment:
do we need to make `emitter_name` a required field? I was thinking that we could generate a unique name if not set.

Reply:
Changed in ea61288.